docs/api/advanced/snapshot.mdx

```diff
@@ -7,22 +7,89 @@ description: 'Create a snapshot of current state'
 
 # snapshot
 
-#### A snapshot takes a proxy and returns a immutable object, unwrapped from the proxy.
+snapshot takes a proxy and returns an immutable object, unwrapped from the proxy.
 
-Immutability is achieved by deeply freezing the object. In sequential snapshot calls, when the values in the proxy have not changed, a pointer to the same object is returned. This allows for shallow comparison in render functions, preventing spurious renders. Snapshots also throw promises, making them work with React Suspense.
+Immutability is achieved by deeply copying & freezing the object. In sequential snapshot calls, when the values in the proxy have not changed, a pointer to the same object is returned. This allows for shallow comparison in render functions, preventing spurious renders. Snapshots also throw promises, making them work with React Suspense.
 
 ```js
 import { proxy, snapshot } from 'valtio'
 
 const store = proxy({ name: 'Mika' })
-const unwrappedStore = snapshot(store) // unwrapped store is the original object, unproxied
+const unwrappedStore = snapshot(store) // unwrapped store is a copy of the original object, unproxied
```

Comment:
Pedantically the snapshot is not literally the `{ name: 'Mika' }` instance passed to `proxy`.

Reply:
What you pass to `proxy()` is "initializing value" and it's different from `store` too.
Again, `snapshot()` return an immutable object which is different from `store`, but it's super optimized copying. I'm afraid by simply saying `a copy of`, it feels like `snapshot()` does deep copying on each call.

Reply:
> What you pass to proxy() is "initializing value" and it's different from store too.

Right, but imo the comment as-is, "unwrapped store is the original object" to me insinuated it is literally the same instance as the initializing value, which it is not.

I've edited a bit (will push the edit soon...) and thrown in an "efficient" to see if that helps? :-)

Reply:
efficient deep copying sounds good!